app/src/main/java/com/codingblocks/cbonlineapp/activities/CourseActivity.kt

```diff
@@ -204,85 +212,100 @@ class CourseActivity : AppCompatActivity(), AnkoLogger {
     }
 
     private fun fetchCourse() {
-        val service = Clients.onlineV2JsonApi
-        Clients.onlineV2JsonApi.courseById(courseId).enqueue(retrofitCallback { t, resp ->
-            resp?.body()?.let { course ->
-                skeletonScreen.hide()
-                fetchInstructors(course.id)
-                batchAdapter = BatchesAdapter(ArrayList(), object : OnCartItemClickListener {
-                    override fun onItemClick(id: String, name: String) {
-                        addtocart(id, name)
+        viewModel.getCourse(object : OnCompleteCourseListener {
+            @Suppress("UNCHECKED_CAST")
+            override fun onCompleteJson(response: Response<JsonObject>) {
```

Comment:
Don't use listeners/callback method use livedata instead see other viewmodels for reference

Reply:
on it